core/interfaces.go

```diff
@@ -119,6 +121,7 @@ type StorageAdder interface {
 	AddCertificate([]byte, int64) (string, error)
 	AddSCTReceipt(SignedCertificateTimestamp) error
 	RevokeAuthorizationsByDomain(AcmeIdentifier) (int64, int64, error)
+	AddNameSet(NameSet) error
```

Comment:
I think there's no need for AddNameSet to be an RPC method. In normal operation, I would expect it to get called internally from AddCertificates in SA. And for the backfill, I think it's preferable to have a copy of that logic rather than create an unneeded RPC. Similarly, I don't think we need GetNameSet. We just need AddCertificate + CountValidNameSets.